### ISSUE TITLE
Crosswalk dedupe logic error

### ISSUE THREAD

**Issue by jason-p-pickering** [pepfar-datim/datim-dedupe#76]
For DSD-TA crosswalk duplicates, the following logic should apply. 

1) By default, the value presented to the user should be the Sum of all of the TA values, minus the DSD value. If this value is negative, it should be zero. As an example

DSD = 10
TA1 = 3
TA2 = 6
Default value = 0

DSD = 10
TA1 = 13
TA2 = 16
TADedupe=0
Default value = 19

DSD = 10
TA1 = 13
TA2 = 16
TADedupe=-13
Default value = 6

2) The user should be allowed to specify a custom value. 

Lower bound = SUM(TA values) - DSD value. If negative, then zero. This represents the maximum extent of possible DSD/TA overlap. 
Upper bound = SUM(TA Values)

Original bug report is here https://datim.zendesk.com/agent/tickets/12377. 

User could not specify what should be a valid value

![image001](https://user-images.githubusercontent.com/2968061/43889199-cc5b3108-9bc3-11e8-94eb-4835e5f837f8.jpg)

**Comment by tomzemp** [comment]
@jason-p-pickering: This seems to be working correctly in the current version of the app (as far as I can observe). I also looked at the code and didn't see any logic errors.

I created an additional TA value on UC for Namibia to test* 

**I was first prompted to resolve the TA values:**

![image](https://user-images.githubusercontent.com/18490902/52956828-8ddeb200-3390-11e9-8d62-e6065d71922e.png)
(range was as expected: max(TA Value) - ∑TA Values)


**After resolving TA, I was prompted to resolve the DSD/TA cross-walk:**
![image](https://user-images.githubusercontent.com/18490902/52956947-d8602e80-3390-11e9-930a-d3054d455921.png)

_The minimum allowed was 33:_ 
Sum of TA Values = 19 + 61 = 80
Minimum = Max(TA Values - DSD Value,0) = Max(80 - 47,0) = 33

_The maximum allowed was 80:_
Max(DSD Value, Sum of TA Values)=Max(47, 19 + 61) = 80

If I tried to enter a value outside of the 33-80 range, the app returned an error:
![image](https://user-images.githubusercontent.com/18490902/52957049-178e7f80-3391-11e9-8e03-3aceb2e1a203.png)

@jason-p-pickering: let me know if you want me to look at some other scenario. Otherwise, I will mark this issue as resolved.

```
INSERT INTO datavalue 
(dataelementid, periodid, sourceid, categoryoptioncomboid, value, storedby, lastupdated, comment, followup, attributeoptioncomboid, created, deleted)
VALUES (
(SELECT dataelementid FROM dataelement WHERE uid='jHjC9XIJbhL'),
(SELECT periodid FROM period WHERE startdate='2018-04-01' AND enddate='2018-06-30'),
(SELECT organisationunitid FROM organisationunit WHERE uid='jzPmH1noExL'),
(SELECT categoryoptioncomboid FROM categoryoptioncombo WHERE uid='H250HduQyXi'),
19,
'tzemp',
(SELECT now()),
'testing dedupe app',
false,
(SELECT categoryoptioncomboid FROM categoryoptioncombo WHERE uid='ITxtmuQ0NVb'),
(SELECT now()),
false
);
```
